cmd/acr-credential-provider/main.go

```diff
@@ -38,27 +41,31 @@ func main() {
 	var RegistryMirrorStr string
 
 	command := &cobra.Command{
-		Use:     "acr-credential-provider configFile",
-		Short:   "Acr credential provider for Kubelet",
-		Long:    `The acr credential provider is responsible for providing ACR credentials for kubelet`,
-		Args:    cobra.MinimumNArgs(1),
-		Version: version.Get().GitVersion,
-		Run: func(_ *cobra.Command, args []string) {
-			if len(args) != 1 {
-				klog.Errorf("Config file is not specified")
-				os.Exit(1)
+		Use:   "acr-credential-provider configFile",
+		Short: "Acr credential provider for Kubelet",
+		Long:  `The acr credential provider is responsible for providing ACR credentials for kubelet`,
+		Args: func(_ *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("Config file is not specified")
 			}
-
+			if len(args) > 1 {
+				return fmt.Errorf("expected exactly one argument (config file); Got arguments: %v", args)
+			}
+			return nil
+		},
+		Version: version.Get().GitVersion,
+		RunE: func(cmd *cobra.Command, args []string) error {
 			acrProvider, err := credentialprovider.NewAcrProviderFromConfig(args[0], RegistryMirrorStr)
 			if err != nil {
 				klog.Errorf("Failed to initialize ACR provider: %v", err)
-				os.Exit(1)
+				return err
```

Comment:
I think returning error will lead the plugin to exiting with non-zero, right?

Reply:
The error will be passed on to the return value of `command.ExecuteContext` and finally causing os.Exit(1), which is equivalent to exiting with non-zero.